File: frontend/src/service/table/Ausgabe.ts
import { z } from 'zod'

import { DateStringSchema } from '../schemas'
import { PositionSchema } from './Bestellung'

export const AusgabeSchema = z.object({
  art: z.literal('ausgabe'),
  id: z.uuid(),
  userId: z.number().int().min(1),
  // Eingefrorener Name des Akteurs, beschriftet den Historien-Eintrag.
  userName: z.string().min(1),
  tischId: z.number().int().min(1),
  positionen: PositionSchema.array().min(1),
  kommentar: z.string().max(100),
  ausgegebenAm: DateStringSchema,
})
export type Ausgabe = z.infer<typeof AusgabeSchema>

File: backend/domain/kasse/ausgabe.go
package kasse

import (
	"time"

	z "github.com/Oudwins/zog"
)

type Ausgabe struct {
	ID           string
	UserID       int
	TischID      int
	Positionen   []Position
	Kommentar    string
	AusgegebenAm time.Time
}

var ausgabeSchema = z.Struct(z.Shape{
	"ID":           z.String().UUID().Required(),
	"UserID":       z.Int().GTE(1).Required(),
	"TischID":      z.Int().GTE(1).Required(),
	"Positionen":   z.Slice(positionSchema).Min(1).Required(),
	"Kommentar":    z.String().Max(100),
	"AusgegebenAm": z.Time().Required(),
})
